### ISSUE TITLE
"Exception during node startup: Is a directory" when using Postgres for Enterprise

### ISSUE THREAD

**Issue by sassit** [corda/corda#6099]
Another issue I ran into while upgrading to Enterprise, the error is not really revealing a lot. It happens when I switch on DB support on DockerForm nodes:

```
 drivers=['lib/postgresql-42.2.8.jar']
        extraConfig = [
//                'database.runMigration' : 'true', // leads to an error, being unknown ...
                'dataSourceProperties.dataSource.user' : 'a',
                'dataSourceProperties.dataSource.password' : 'b',
                'dataSourceProperties.dataSource.url' : 'jdbc:postgresql://party-a-db:5432/postgres',
                'dataSourceProperties.dataSourceClassName' : 'org.postgresql.ds.PGSimpleDataSource'
        ]
```

```
notary_1                      | ! ATTENTION: This node is running in development mode!  This is not safe for production deployment.
partya_1                      | ! ATTENTION: This node is running in development mode!  This is not safe for production deployment.
partyb_1                      | ! ATTENTION: This node is running in development mode!  This is not safe for production deployment.
notary_1                      | [ERROR] 23:11:09+0000 [main] internal.NodeStartupLogging. - Exception during node startup: Is a directory [errorCode=1nl9meo, moreInformationAt=https://errors.corda.net/OS/4.3/1nl9meo]
partyb_1                      | [ERROR] 23:11:10+0000 [main] internal.NodeStartupLogging. - Exception during node startup: Is a directory [errorCode=1nl9meo, moreInformationAt=https://errors.corda.net/OS/4.3/1nl9meo]
partya_1                      | [ERROR] 23:11:10+0000 [main] internal.NodeStartupLogging. - Exception during node startup: Is a directory [errorCode=1nl9meo, moreInformationAt=https://errors.corda.net/OS/4.3/1nl9meo]
```

**Comment by sassit** [comment]
Closing it for now, even though Enterprise is referenced open source is started in docker-compose. Seems to be a different problem.

**Comment by michaeldowling** [comment]
@sassit I'm encountering this issue right now, after configuring a 4 node networking utilizing "corda-tools-network-bootstrapper-4.4.jar".  My various _node.conf files are right from example:

`
devMode=true
myLegalName="O=AlphaBank,L=Seattle,C=US"
p2pAddress="alphabank:10200"
rpcSettings {
    address="0.0.0.0:10201"
    adminAddress="0.0.0.0:10202"
}
security {
    authService {
        dataSource {
            type=INMEMORY
            users=[
                {
                    password=test
                    permissions=[
                        ALL
                    ]
                    user=test
                }
            ]
        }
    }
}
sshd {
    port = 2222
}
`

When I run docker-compose up, I get the following error:
[ERROR] 20:45:23+0000 [main] internal.NodeStartupLogging. - Exception during node startup: Is a directory [errorCode=1nl9meo, moreInformationAt=https://errors.corda.net/OS/4.4/1nl9meo]

The moreInfoAt url doesn't seem to exist.  My hunch is it has something to do with the node's persistence using H2?

Did you resolve this at all on your end?  What am i missing?

**Comment by jweate** [comment]
@michaeldowling I had a similar issue when running docker-compose up, I was able to track down the stack trace in the node.log file in the logs volume where I saw the NetworkParametersReader was being called. Eventually I realized my local network-parameters did not exist so compose was creating an empty directory for me, which was then causing it to fail. Creating the network-params with the network bootstrapper and properly mounting fixed the issue. Hope this helps!
